### DISCUSSION TITLE
Questions regarding Managed PostgreSQL service

### DISCUSSION THREAD

**Discussion by MrOctopus** [ubicloud/ubicloud#2340] [General]
Hi,

I've been reading up on your managed PostgreSQL service, and it looks very promising. However, there are (potentially) two key blockers that can prevent me from taking advantage of it, which I'd very much want have clarified first.

1. Does the service support only whitelisting a certain ip-range? Exposing the database to the internet and relying purely on the credentials are a no-go for me as far as security is concerned.
2. Seeing as you are an American company: for your European customers is it possible to have a Data Processing Agreement (DPA) signed, and are there any plans to enter into the Data Privacy Framework (https://www.dataprivacyframework.gov/)? Asking here in terms of staying compliant within the boundaries of GDPR.

**Comment by byucesoy** [comment]
Hi @MrOctopus.

1. Yes, you can set firewall rules to allow only certain ip address or range for accessing to the database. We are also working on a feature that allows communicating with your database from private network if your VM is also hosted in Ubicloud. In such cases, you can block public network communication entirely.
2. We contract with our customers based in the European Economic Area through Ubicloud B.V., a Dutch entity, and we host all of the content data within servers in the EU. We also offer DPA to our customers based in EEA and in our EU region, if they plan to store personal data in the database.  We are in the process of updating our website to reflect this. We do not have any plans to enter into the privacy framework at the moment; however, we rely on EU's Standard Contractual Clauses to comply with GDPR's international transfer requirements.
In all cases, when you provision the database in one of our EU regions, your data stays within the EU. This includes all the data in your database, backups, and your login credentials. We also encrypt the data both at rest and in-transit. For data at rest, we use a technique called envelope encryption. This ensures that compromising either the database host or the key repository alone isn’t sufficient to decrypt the data.

I hope these answers are helpful. Please let me know if you have any other questions.

**Comment by MrOctopus** [comment]
Thanks, that's very helpful. As for the following:
> We are also working on a feature that allows communicating with your database from private network if your VM is also hosted in Ubicloud.

Is there any ETA on when this will be available to customers?

**Reply by byucesoy** [reply]
We have [a draft PR](https://github.com/ubicloud/ubicloud/pull/2290/files), but we don't have a concrete timeline at the moment.
